applications: nrf_desktop: Improve DFU over SMP documentation (#11240)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/ble_latency.rst

The module listens for the following events related to data transfer initiated by the connected Bluetooth central:

* ``config_event`` - This event is received when the :ref:`nrf_desktop_config_channel` is in use.
* ``ble_smp_transfer_event`` - This event is received when the firmware update is received either by :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.

When these events are received, the module sets the connection latency to low.

> When the :ref:`nrf_desktop_config_channel` is no longer in use and the firmware update is not received by :ref:`nrf_desktop_ble_smp` nor :ref:`nrf_desktop_dfu_mcumgr` (no mentioned events for ``LOW_LATENCY_CHECK_PERIOD_MS``), the module sets the connection latency to :kconfig:option:`CONFIG_BT_PERIPHERAL_PREF_LATENCY` to reduce the power consumption.

Reviewer: ```suggestion
When the :ref:`nrf_desktop_config_channel` is no longer in use, and neither :ref:`nrf_desktop_ble_smp` nor :ref:`nrf_desktop_dfu_mcumgr` receive firmware updates (no mentioned events for ``LOW_LATENCY_CHECK_PERIOD_MS``), the module sets the connection latency to :kconfig:option:`CONFIG_BT_PERIPHERAL_PREF_LATENCY` to reduce the power consumption.
```
